JustSaying.AwsTools/SqsNotificationListener.cs

```diff
@@ -269,14 +260,53 @@ public void ProcessMessageAction(Amazon.SQS.Model.Message message)
                     message,
                     ex.StackTrace);
                 Log.Error(ex, msg);
+
                 if (typedMessage != null)
                 {
                     _messagingMonitor.HandleException(typedMessage.GetType().Name);
                 }
+
                 _onError(ex, message);
-                
             }
         }
+
+        private bool CallMessageHandlers(Message message)
+        {
+            List<HandlerFunc> handlerFuncs;
+            var foundHandlers = _handlers.TryGetValue(message.GetType(), out handlerFuncs);
+
+            if (!foundHandlers)
+            {
+                return true;
+            }
+
+            var allHandlersSucceeded = true;
+            foreach (var handlerFunc in handlerFuncs)
+            {
+                var watch = new System.Diagnostics.Stopwatch();
+                watch.Start();
+
+                var thisHandlerSucceeded = handlerFunc(message);
+                allHandlersSucceeded = allHandlersSucceeded && thisHandlerSucceeded;

```

Comment:
The multiple handlers for a message scenario is not very common but it's certainly possible. 
Here is my thoughts on the questions you've raised.

You should not delete a message unless all handlers of that message return success otherwise the reliability of the library will be compromised. 

Execution of handlers could be parallelised but the decision whether to delete or keep the message can only be made after all handlers of a message have returned.

Simply deleting messages that have no handlers is ignoring a misconfiguration and is not correct. But I would address that separately and in isolation in case some users are relying on this behaviour.